src/AggregateFunctions/AggregateFunctionUniq.h

```diff
@@ -408,6 +409,17 @@ class AggregateFunctionUniq final : public IAggregateFunctionDataHelper<Data, Ag
         detail::Adder<T, Data>::add(this->data(place), columns, num_args, row_begin, row_end, flags, null_map);
     }
 
+    bool isParallelizeMergePrepareNeeded() const override { return true;}
+
+    void parallelizeMergePrepare(AggregateDataPtrs & places, ThreadPool & thread_pool) const override
+    {
+        std::vector<DataSet *> data_vec;
```

Comment:
```
data_vec.resize(places.size());
```

Reply:
I have tried to add the code line `data_vec.resize(places.size());` after the initialization of `std::vector<DataSet *> data_vec;` But the ClickHouse server crashed. I will check that.

Reply:
Add the resize() for data_vec in parallelizeMergePrepare.